app/main.py

```diff
@@ -1,3 +1,65 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    # def __add__(self, other: ("Distance", int, float)) -> "Distance":
+    def __add__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
+
```

Comment:
When I said that the else is redundant, I meant this:
```
...
elif isinstance(other, (int, float)):
        return Distance(self.km + other)
raise TypeError(
        f"Unsupported operand type(s) for +: "
        f"'Distance' and {type(other)}"
  )
```

In case you don't want to raise a TypeError:
```
if isinstance(other, Distance):
    return Distance(self.km + other.km)
return Distance(self.km + other)
```

Reply:
Thank you for your comments. Here are my corrections if I've got it right.